src/addr.rs

```diff
@@ -322,6 +326,44 @@ impl Sub<VirtAddr> for VirtAddr {
     }
 }
 
+#[cfg(feature = "step_trait")]
+impl Step for VirtAddr {
+    fn steps_between(start: &Self, end: &Self) -> Option<usize> {
+        let mut steps = end.0.checked_sub(start.0)?;
+
+        // Check if we jumped the gap.
+        if end.0.get_bit(47) && !start.0.get_bit(47) {
+            steps -= 0xffff_0000_0000_0000;
+        }
+
+        usize::try_from(steps).ok()
+    }
+
+    fn forward_checked(start: Self, count: usize) -> Option<Self> {
+        let offset = u64::try_from(count).ok()?;
+        let mut addr = start.0.checked_add(offset)?;
+
+        // Jump the gap by sign extending the 47th bit.
+        if addr.get_bits(47..) == 0x1 {
```

Comment:
What happens if `count` is so large that this `get_bits` returns 0x3 or more? We should probably return `None`, right?

Reply:
> What happens if `count` is so large that this `get_bits` returns 0x3 or more? We should probably return `None`, right?

Good point. Note that we can also get a larger value (`0x1ffff`) if `start` is in the upper half. We can check if `offset` is bigger than the total address space and return `None` if that's the case.